### PR TITLE
fido2-manage: init at 0-unstable-2024-09-24

### PR DESCRIPTION
**Pull request by Srylax**
Add `fido2-manage`, allows to manage fido2 keys. Developed by Token2

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable: `Since upstream has no tests and a hardware key would be required this is hard to do`
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes) 
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/fi/fido2-manage/package.nix [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    ++ lib.optionals stdenv.isLinux [
+      xterm
+      udev
+      pcsclite
+    ]
+    ++ lib.optionals stdenv.isDarwin [
+      libuv
+      libsolv
+      libcouchbase
+      gnugrep
```
Comment (SuperSandro2000) [inline pkgs/by-name/fi/fido2-manage/package.nix:64]: grep as build input? Shouldn't this go into native build inputs?

File: pkgs/by-name/fi/fido2-manage/package.nix [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        --replace-fail "\''${prefix}/@CMAKE_INSTALL_LIBDIR@" "@CMAKE_INSTALL_FULL_LIBDIR@"
+    ''
+    + lib.optionalString stdenv.isDarwin ''
+      substituteInPlace ./CMakeLists.txt \
+        --replace-fail "/\''${CMAKE_INSTALL_LIBDIR}" "/lib"
+    '';
+
+  postInstall =
+    lib.optionalString stdenv.isLinux ''
+      install $src/fido2-manage.sh $out/bin/${pname}
```
Comment (SuperSandro2000) [inline pkgs/by-name/fi/fido2-manage/package.nix:83]: Please don't substitute pname for the binary path, just hardcode the string instead

File: pkgs/by-name/fi/fido2-manage/package.nix [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+      install $src/gui.py $out/bin/${pname}-gui
+    ''
+    + lib.optionalString stdenv.isDarwin ''
+      install $src/fido2-manage-mac.sh $out/bin/${pname}
+    '';
+
+  desktopItems = lib.optionals stdenv.isLinux [
+    (makeDesktopItem rec {
+      desktopName = "Fido2 Manager";
+      name = pname;
```
Comment (SuperSandro2000) [inline pkgs/by-name/fi/fido2-manage/package.nix:94]: Same here

File: pkgs/by-name/fi/fido2-manage/token2.png
Comment (SuperSandro2000) [inline pkgs/by-name/fi/fido2-manage/token2.png:1]: Please fetch that from the website or so instead

**Review by SuperSandro2000 — DISMISSED** [review]

File: pkgs/by-name/fi/fido2-manage/package.nix (hunk 1 above)
Comment (Srylax) [reply pkgs/by-name/fi/fido2-manage/package.nix]: my bad. removed it

File: pkgs/by-name/fi/fido2-manage/package.nix (hunk 2 above)
Comment (Srylax) [reply pkgs/by-name/fi/fido2-manage/package.nix]: I removed all pname references. Just for my understanding: what is the reasoning behind this? Evaluation time?

**Comment by Srylax** [comment]
@SuperSandro2000 what is the general sentiment on using `with`? Fore example is it preferred to use `with lib.maintainers; [ Srylax ]` or `[ lib.maintainers.Srylax ]`?

**Comment by SuperSandro2000** [comment]
Using with over a small area of code is fine, over the entire file or 20+ lines not.

**Comment by Srylax** [comment]
@SuperSandro2000 are there still some unresolved changes, or can this be merged?

**Pull request merged**